velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -207,6 +207,57 @@ struct UnixTimestampParseWithFormatFunction
   bool invalidFormat_{false};
 };
 
+// Parses unix time in seconds to a formatted string.
+template <typename T>
+struct FromUnixtimeFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<int64_t>* /*unixtime*/,
+      const arg_type<Varchar>* format) {
+    sessionTimeZone_ = getTimeZoneFromConfig(config);
+    if (format != nullptr) {
+      setFormatter(*format);
+      isConstantTimeFormat_ = true;
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<int64_t> second,
+      const arg_type<Varchar> format) {
+    if (!isConstantTimeFormat_) {
+      setFormatter(format);
+    }
+    const Timestamp timestamp{second, 0};
+    result.reserve(maxResultSize_);
+    int32_t resultSize;
+    try {
+      resultSize = formatter_->format(
+          timestamp, sessionTimeZone_, maxResultSize_, result.data());
+    } catch (const VeloxUserError& e) {
```

Comment:
throwing exceptions is expensive; is there a way to avoid these?

Reply:
@mbasmanova, I note Pedro will introduce Status/Result object to replace throwing exception. See https://github.com/facebookincubator/velox/pull/8084#issue-2045401962. Can I fix this issue for all code related to datetime parsing in a separate PR based on Pedro's work?

Reply:
This work is months away from being complete. Throwing and catching exceptions per-row like this has been shown to produce extremely slow queries with performance 10x or more slower than Java.